### ISSUE TITLE
Introduce a command exec mode

### ISSUE THREAD

**Issue by ekkinox** [aandrew-me/tgpt#10]
Hey, nice project.

I have a very similar one: https://github.com/ekkinox/yo, also in Go.

IMO a big feature for an AI TUI is to offer to run commands from user prompt, I suggest you do the same in your app, it'll bring it a lot more value :) 

Also, I suggest you check early how to integrate tests, lint and [goreleaser](https://goreleaser.com/), since you get traction from YouTube, it'll help you in your next iterations ;)

**Comment by aandrew-me** [comment]
Hey thanks for your words!

You have done a great job with your project. I honestly don't know how to run commands from user prompt. 

And thanks for your suggestions. I am new to go and I am loving it.

**Comment by ekkinox** [comment]
You can look at `os/exec`: https://pkg.go.dev/os/exec

It allows you to run commands fro go, and you can use AI to generate commands from user prompt, then run them with go :)

**Comment by aandrew-me** [comment]
Yes I know about `os/exec`

However, I don't know how to parse user commands in such a way to that AI only sends code for execution

**Comment by ekkinox** [comment]
You can check how I did in Yo source code 😉

Basically I ask ChatGPT a JSON containing command and explanation, i just parse it and run it when possible

**Comment by aandrew-me** [comment]
Well the problem is my program is not communicating with OpenAI api.

**Comment by ekkinox** [comment]
I'm pretty sure you can get similar results

**Comment by aandrew-me** [comment]
Well I will see what I can do

**Comment by Bellisario** [comment]
Just through on the same functionality and got a way to proceed...

Running the command "How to rename a folder 'hi to 'hello'? Output only the command needed. Do not explain anything." will only output the command needed, without any explanation.

By "asking" (imposing) the AI to output only the command, then it can be showed to the user that will decide if he wants to run it or not.

I hope this helps!

P.S.: awesome project!

**Comment by ekkinox** [comment]
This was my first approach for Yo, but sometimes AI still adds some text around. To fix it, I asked a JSON to format all this, without explanation, and it works quite well

**Comment by ekkinox** [comment]
Especially with GPT4, even better than 3.5 turbo

**Comment by aandrew-me** [comment]
> This was my first approach for Yo, but sometimes AI still adds some text around. To fix it, I asked a JSON to format all this, without explanation, and it works quite well
> 
> See https://github.com/ekkinox/yo/blob/main/ai/engine.go#L308

So based on prompt mode you are sending text to gpt even before the user writes anything!?

**Comment by ekkinox** [comment]
Exactly, with the system role. You can still regen free keys on your OpenAI profile no ?

**Comment by aandrew-me** [comment]
I understand, you have done a lot of work there.

And sadly no, looks like its just free trial which has already expired.

**Comment by aandrew-me** [comment]
> Exactly, with the system role. You can still regen free keys on your OpenAI profile no ?

You have done an absolutely brilliant job!

**Comment by ekkinox** [comment]
Thx ! You too 👍

**Comment by aandrew-me** [comment]
> Thx ! You too +1

Hey man how are you doing?

I am being faced with a strange problem and I can't understand why its happening. Your help would be so much appreciated.

For some IP addresses when I am sending a request in tgpt, the server is returning 403 error. When I tried with a VPN, I also got that error. But the thing is, when I add a local Proxy to the program, `proxyUrl, _ := url.Parse("http://127.0.0.1:8080")`, the error goes away. Also it works fine with cURL. But for some reason in Go, I am getting an error.
